cloudinit/config/cc_lxd.py

```diff
@@ -126,7 +131,36 @@ def handle(name, cfg, cloud, log, args):
             "storage_pool",
             "trust_password",
         )
+
         subp.subp(["lxd", "waitready", "--timeout=300"])
+
+        # Bug https://bugs.launchpad.net/ubuntu/+source/linux-kvm/+bug/1982780
+        if (
+            cloud.distro.name == "ubuntu"
+            and init_cfg["storage_backend"] == "lvm"
+        ):
+            log.warning(
```

Comment:
This issue isn't limited to ubuntu, it's possible that this is a problem on any given OS/kernel if the right kernel modules are not included. 

I think we instead want to perform a check to see if the dm-thin-pool module is present for the active kernel.

We can either do that through:
1. file test on required dm-thin-pools module
```python
kernel = util.system_info()["uname"][2])
if not os.path.exists(f"/lib/modules/{kernel}/kernel/drivers/md/dm-thin-pool.ko):
   ...log.warn and setup w/ lvm.use_thinpool=False....
```
2 subp.subp(['modinfo', 'dm-thin-pool']) and check if we hit ProcesseExecutionError "modinfo: ERROR: Module dm-thin-pools not found.". 


I'd lean toward solution #1 because solution #2 requires either a strict package dependency on `kmod` or optional runtime package dependency on kmow which I don't think we should grow broadly in cloud-init for this one off use-case when all we need it for is a file check.

Reply:
> I think we instead want to perform a check to see if the dm-thin-pool module is present for the active kernel.

This approach causes a behavioral change that would be unpleasant to debug: kmod package update will change cloud-init behavior under this scheme. However, since we're logging this when the kmod is missing, I guess there is a breadcrumb.

I don't feel strongly about this either way and don't have a better suggestion. I'll grab the first option.

Thanks!

Reply:
> kernel = util.system_info()["uname"][2])

Note to self: If we get rid of the cast-to-list, we get a named tuple for free (and more readable code).

Reply:
+1 I was thinking the same, though we'd maybe have to look at how that is represented in `/run/cloud-init/isntance-data.json` to be sure we aren't doing some funky object-style annotations in the JSON output as I believe this data is surfaced there too.